### PR TITLE
feat(e-z): add presence

### PR DESCRIPTION
**Pull request by tsoxas**
## Description 
Adding presence for e-z.host and e-z.bio

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>
image
image
image

</details>

### CODE REVIEW

File: websites/E/e-z/presence.ts [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+		uid = document.querySelector("p[id=premid-uid]")?.textContent;
+
+	switch (document.location.hostname) {
+		case "e-z.host":
+			presenceData.buttons = [
+				{
+					label: "e-z.host",
+					url: "https://e-z.host",
+				},
+			];
```
Comment (theusaf) [inline websites/E/e-z/presence.ts:19]: buttons to homepages are [not allowed](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md#presencets).
```suggestion
```

File: websites/E/e-z/presence.ts [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+		case "e-z.bio":
+			if (document.location.pathname === "/") {
+				presenceData.details = "e-z.bio front page";
+				presenceData.buttons = [
+					{
+						label: "e-z.bio",
+						url: "https://e-z.bio",
+					},
+				];
```
Comment (theusaf) [inline websites/E/e-z/presence.ts:93]: buttons to homepages are [not allowed](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md#presencets).
```suggestion
```

File: websites/E/e-z/presence.ts [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+			} else {
+				presenceData.details = `Viewing ${document.location.pathname.replace(
+					"/",
+					""
+				)}'s bio page`;
+				presenceData.buttons = [
+					{
+						label: "e-z.bio",
+						url: "https://e-z.bio",
+					},
```
Comment (theusaf) [inline websites/E/e-z/presence.ts:103]: buttons to homepages are [not allowed](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md#presencets).
```suggestion
```

File: websites/E/e-z/dist/metadata.json [hunk 4]
```diff
@@ -0,0 +1,25 @@
+{
+	"$schema": "https://schemas.premid.app/metadata/1.7",
+	"author": {
+		"name": "tsoxas",
+		"id": "787787652644601857"
+	},
+	"service": "E-Z",
```
Comment (theusaf) [inline websites/E/e-z/dist/metadata.json:7]: Rename the folder. Helpful [Stackoverflow question](https://stackoverflow.com/questions/6899582/i-change-the-capitalization-of-a-directory-and-git-doesnt-seem-to-pick-up-on-it).

File: websites/E/e-z/dist/metadata.json [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+	"description": {
+		"en": "E-Z is the home of e-z.host. e-z.host is a ShareX image uploader with many unique features and tons of domains."
+	},
+	"url": [
+		"e-z.host",
+		"e-z.bio"
+	],
+	"version": "1.0.0",
+	"logo": "https://cdn.e-z.host/e-zimagehosting/ez256.png",
+	"thumbnail": "https://cdn.e-z.host/e-zimagehosting/e-z_host_banner.png",
```
Comment (theusaf) [inline websites/E/e-z/dist/metadata.json:17]: Use Imgur. Also, logos must have a [minimum resolution of 512x512px](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md#logo).

**Review by theusaf — CHANGES_REQUESTED** [review]

File: websites/E/e-z/presence.ts [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+					{
+						label: "View Page",
+						url: `https://e-z.bio${document.location.pathname}`,
```
Comment (DarkVillager) [inline websites/E/e-z/presence.ts:106]:
```suggestion
						url: document.location.href,
```

**Review by DarkVillager — CHANGES_REQUESTED** [review]

File: websites/E/e-z/presence.ts [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+});
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: "https://cdn.e-z.host/e-zimagehosting/ez256.png",
+		},
+		username = document.querySelector("p[id=premid-username]")?.textContent,
+		uid = document.querySelector("p[id=premid-uid]")?.textContent;
+
+	switch (document.location.hostname) {
```
Comment (DarkVillager) [inline websites/E/e-z/presence.ts:12]: Use variables for this.
[Example of the right variables.](https://github.com/PreMiD/Presences/commit/958d5b80733887427f412471048b9614390e1fb7#diff-adb2cc91e493add5ee896439dcca4f8073c1cda9b686816134ca8910eba6c8acR24)

**Review by DarkVillager — CHANGES_REQUESTED** [review]

**Comment by DarkVillager** [comment]
@tsoxas mark changes as resolved when you’ve done them

**Comment by tsoxas** [comment]
> @tsoxas mark changes as resolved when you’ve done them 

Done. Let me know if there are any more changes needed

File: websites/E/e-z/presence.ts [hunk 8]
```diff
@@ -0,0 +1,118 @@
+const presence = new Presence({
+	clientId: "1018191690210754693",
+});
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: "https://cdn.e-z.host/e-zimagehosting/ez256.png",
```
Comment (DarkVillager) [inline websites/E/e-z/presence.ts:7]:
```suggestion
			largeImageKey: "https://i.imgur.com/Crfg6sn.png",
```

**Review by DarkVillager — CHANGES_REQUESTED** [review]

File: websites/E/e-z/presence.ts [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+
+			switch (pathname) {
+				case "/":
+					presenceData.state = "Viewing the home page";
+					presenceData.buttons = [
+						{
+							label: "Discord",
+							url: "https://discord.gg/ez",
+						},
+					];
```
Comment (EncryptedDev) [inline websites/E/e-z/presence.ts:33]: This is forbidden 
image

File: websites/E/e-z/presence.ts [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+	const presenceData: PresenceData = {
+			largeImageKey: "https://i.imgur.com/Crfg6sn.png",
+		},
+		username = document.querySelector("p[id=premid-username]")?.textContent,
+		uid = document.querySelector("p[id=premid-uid]")?.textContent,
+		{ pathname, href, hostname } = document.location;
+
+	switch (hostname) {
+		case "e-z.host":
+			if (typeof username !== "undefined" && typeof uid !== "undefined") {
```
Comment (EncryptedDev) [inline websites/E/e-z/presence.ts:15]:
```suggestion
			if (username && uid) {
```

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

**Comment by ignSKRRRTT** [comment]
"1 review requesting changes by reviewers with write access" but it doesn't say which check it is. All the checks have been resolved

**Review by EncryptedDev — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Comment by EncryptedDev** [comment]
Resolve CI so this can be merged

**Comment by tsoxas** [comment]
@EncryptedDev I fixed the checks failing

**Pull request merged**